depends/funcs.mk

```diff
@@ -157,48 +158,60 @@ $(1)_autoconf += LDFLAGS="$$($(1)_ldflags)"
 endif
 endef
 
+shortdate="+%H:%M:%S"
+
 define int_add_cmds
-$($(1)_fetched):
+$($(1)_introduced):
+	$(AT)echo "============================================================"
+	$(AT)echo "   Package : $(1)"
+	$(AT)echo "   Date    : `date`"
+	$(AT)echo "============================================================"
+	$(AT)mkdir -p $$(@D)
+	$(AT)touch $$@
+$($(1)_fetched): | $($(1)_introduced)
 	$(AT)mkdir -p $$(@D) $(SOURCES_PATH)
 	$(AT)rm -f $$@
 	$(AT)touch $$@
-	$(AT)cd $$(@D); $(call $(1)_fetch_cmds,$(1))
+	$(AT)cd $$(@D); ( echo -n -e ""; $(call $(1)_fetch_cmds,$(1)) ) > .fetch_output 2>&1 || ( cat .fetch_output && false )
```

Comment:
Instead of .fetch_input, let's use ```$$($(1)_download_dir)/.fetch_output```

That way we're guaranteed to never have any weird races or overwrites.

It'll also take care of cleaning up the logs when successful.